en/sensor/pmw3901.md

```diff
@@ -0,0 +1,148 @@
+This board was mainly tested with the [Lidar Lite V3](../sensor/lidar_lite.md) on larger vehicles and with the [Lanbao CM8JL65](../sensor/cm8jl65_ir_distance_sensor.md) on smaller vehicles.
+
+The range needed is dependent on the application.
+When flying indoors, generally 4 meters height are enough.
+If optical flow is also needed outdoors to support the position control in environments where GPS might have issues, then more range is advised (10 meters or more).
+
+The PMW3901 driver was written to be directly plugged into the SPI port on the Pixhawk 4 using the chip select 1.
+The support for this driver on other boards can then easily be extended to other boards by modifying the `SPI_BUS` and the `SPI_DEV` in the [PMW3901 driver](https://github.com/PX4/Firmware/tree/{{ book.px4_version}}/src/drivers/optical_flow/pmw3901).
+
+The following diagram shows how to wire the sensor to a Pixhawk 4.
+
+![Bitcraze PH4 Pinout](../../assets/hardware/sensors/pmw3901/ph4-bitcraze-flow-pinout.png)
+
+When plugged into the SPI port of the Pixhawk 4, the Bitcraze flow module will be automatically detected at bootup.
+No parameters will have to configured, unless you decide to change the [orientation of the sensor](#orientation).
+
+### Tindie
+
+The Tindie optical flow module exposes the SPI interface from the PMW3901 module exactly as on the Bitcraze module.
+The only difference here is the size of the board and the fact that this module doesn't have a distance sensor onboard.
+Unlike the Bitcraze module where you could try to use the onboard distance sensor, with this one it is mandatory to use one of the [distance sensors](../sensor/rangefinders.md) supported by PX4.
+
+![Tindie PH4 Pinout](../../assets/hardware/sensors/pmw3901/ph4-tindie-flow-pinout.png)
+
+### Hex
+
+The Hex optical flow module is a very small board containing the PMW3901 flow module, the VL53L1X distance sensor and an IMU to better synchronize the flow data with the gyro data.
+These three sensors are then sampled by a microcontroller onboard.
+The goal of the microcontroller is to then expose 2 UAVCAN messages containing all the necessary data.
+
+This board can be connected to the CAN port on any Pixhawk board.
+All tests were performed connecting it to the CAN1 port on a Pixhawk 4.
+Differently from the other sensors, this one needs UAVCAN running on PX4.
+To enable UAVCAN you will have to set the UAVCAN_ENABLE parameter to 2 (if you have have UAVCAN sensors but no motor controllers) or to 3 (if you have also your motor controllers on CAN).
+
+As for the other optical flow boards, it is recommended to use an external distance sensor instead of the onboard one due to the limited range of the VL53L1X.
+
+![Hex PH4 Pinout](../../assets/hardware/sensors/pmw3901/ph4-hex-optical-flow.png)
+TODO
+
+Regarding rotations, also this board will have a default orientation and if necessary this could be modified through a parameter.
+
+### Thone
+WORK IN PROGRESS
```

Comment:
This says UART, so presumably you need to set up the serial parameters? What config parameter do you use for the driver?

Reply:
This is still an open pr: https://github.com/PX4/Firmware/pull/12148. The parameter to set will be SENS_TFLOW_CFG to select which telemetry port to use.

Reply:
Marked as in-progress with link to PR added: https://github.com/PX4/px4_user_guide/commit/21c438ae5d19982c9cc302b5d2bc212d11144eb0